src/test/functional/Superescalar/loop.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { expect, beforeEach, test } from 'vitest'
+import { Code } from '../../../core/Common/Code';
+import { Superescalar } from '../../../core/Superescalar/Superescalar';
+import { SuperescalarStatus } from '../../../core/Superescalar/SuperescalarEnums';
+import { codeInput, vecContent, sumContent, resultContent } from "../code/bucle";
+import { codeInput as doubleCodeInput } from "../code/bucledoble";
+import { codeInput as softCodeInput } from "../code/buclesoft";
+
+
+const context: { code: Code, machine: Superescalar } = { code: null, machine: null };
+
+beforeEach(() => {
+    context.code = new Code();
+    context.machine = new Superescalar();
+    context.machine.init(true);
+});
+
+test('Bucle.pla is executed properly', t => {
+    // Load code
+    context.code.load(codeInput);
+    context.machine.code = context.code;
+
+    // Load memory
+    context.machine.memory.setDatum(40, sumContent);
+    for (let i = 50; i < vecContent.length + 50; i++) {
+        context.machine.memory.setDatum(i, vecContent[i - 50]);
+    }
+
+    // Execute code
+    while (context.machine.tic() !== SuperescalarStatus.SUPER_ENDEXE) { }
+
+
+    // Check where the program counter is
+    expect(context.machine.pc).toBe(11);
+
+    // Check the result
+    let result = Array.from(Array(resultContent.length).keys()).map(x => context.machine.memory.getDatum(70 + x).datum);
```

Comment:
Can we use this instead for all cases? Sorry for changing my mind here again :sweat_smile: 

```
    // Check the result
    const resultBaseAddress = 70;
    const result = context.machine.memory.data.slice(
        resultBaseAddress, resultBaseAddress + resultContent.length
    );

    expect(result).toStrictEqual(resultContent);

```

Reply:
fixing in the next commit